### ISSUE TITLE
[react-jss] Observable rules/values

### ISSUE THREAD

**Issue by HenriBeck** [cssinjs/jss#995]
__Expected behavior:__
Observables should work without explicitly passing `link: true` as options and should be part of the static stylesheet. And no dynamic stylesheet should be generated when only observables are used.

__Describe the bug:__
Currently, observables are part of the static sheet but the user needs to pass `link: true` and a dynamic sheet will also be generated.

__Codesandbox link:__ 
https://codesandbox.io/s/01ro5xn32p

__Versions (please complete the following information):__
 - jss: 10.0.0-alpha.8

**Comment by kof** [comment]
Why that expectation? The plan was to treat observables same  way as fn values reg. link option.

**Comment by kof** [comment]
Ah in react-jss we should be able to detect observables like we do with fn values and set link: true automatically

**Comment by HenriBeck** [comment]
Yeah, also observables can go into the static sheet and are actually currently only in the static sheet

**Comment by kof** [comment]
Oh noo, observables should go into a dynamic sheet!!!!!

**Comment by HenriBeck** [comment]
But observables are shared between all mounted components.

**Comment by kof** [comment]
Right, I forget that we currently have only 2 ways to create different observables:
1. Use a function that returns styles with component together and a accepts/creates Observables
1. Use theming as a mechanism to create a new observable

**Comment by kof** [comment]
Mb to support an Observable per element, we should implement function value that can return an observable, so that Observable instance can come from the props/data.

**Comment by kof** [comment]
Since I forget this every time myself, we should document this current behaviour
